### PR TITLE
Add did-you-mean functionality for RelationUnknown errors

### PR DESCRIPTION
**Pull request by mfussenegger**
## Checklist

 - [x] User relevant changes are recorded in ``CHANGES.txt``
 - [x] Touched code is covered by tests
 - [x] Documentation has been updated if necessary
 - [x] [CLA](https://crate.io/community/contribute/cla/) is signed
 - [x] This does not contain breaking changes, or if it does:
    - It is released within a major release
    - It is recorded in ``CHANGES.txt``
    - It was marked as deprecated in an earlier release if possible
    - You've thought about the consequences and other components are adapted
      (E.g. AdminUI)

### CODE REVIEW

File: sql/src/main/java/io/crate/metadata/Schemas.java [hunk 1]
```diff
@@ -122,29 +122,49 @@ public TableInfo resolveTableInfo(QualifiedName ident, Operation operation, User
...
+        ArrayList<String> candidates = new ArrayList<>();
+        for (TableInfo table : tables) {
+            if (user.hasAnyPrivilege(Privilege.Clazz.TABLE, table.ident().fqn())) {
+                String candidate = table.ident().name();
+                if (levenshteinDistance.getDistance(tableName, candidate) > 0.7f || tableName.equalsIgnoreCase(candidate)) {
+                    candidates.add(candidate);
+                }
+            }
+        }
+        return candidates;
```
Comment (mfussenegger) [inline sql/src/main/java/io/crate/metadata/Schemas.java]: Do you think it is worth to include some scoring for ordering and limit this to maybe max 5 candidates?

Comment (kovrus) [reply sql/src/main/java/io/crate/metadata/Schemas.java]: sounds nice! so users are not overwhelmed with hints.

Shall we also add a change log entry for both this and schema hints?

Comment (mfussenegger) [reply sql/src/main/java/io/crate/metadata/Schemas.java]: @kovrus pushed a fixup

File: sql/src/main/java/io/crate/metadata/Schemas.java [hunk 2]
```diff
@@ -122,36 +125,77 @@ public TableInfo resolveTableInfo(QualifiedName ident, Operation operation, User
...
-                    candidates.add(availableSchema);
+                float score = levenshteinDistance.getDistance(schema.toLowerCase(Locale.ENGLISH), availableSchema.toLowerCase(Locale.ENGLISH));
+                if (score > 0.7f) {
+                    candidates.add(new Candidate(score, availableSchema));
                 }
             }
         }
-        return candidates;
+        candidates.sort(Comparator.comparing((Candidate x) -> x.score).reversed());
+        return candidates.stream()
```
Comment (kovrus) [inline sql/src/main/java/io/crate/metadata/Schemas.java]: we could have use a priority queue as well, but it is probably not that critical 👍

**Pull request merged**